future/atomicTransactionComposer.go

```diff
@@ -0,0 +1,681 @@
+package future
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/algorand/go-algorand-sdk/abi"
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand-sdk/crypto"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+/**
+ * This type represents a function which can sign transactions from an atomic transaction group.
+ * @param txnGroup - The atomic group containing transactions to be signed
+ * @param indexesToSign - An array of indexes in the atomic transaction group that should be signed
+ * @returns A promise which resolves an array of encoded signed transactions. The length of the
+ *   array will be the same as the length of indexesToSign, and each index i in the array
+ *   corresponds to the signed transaction from txnGroup[indexesToSign[i]]
+ */
+type TransactionSigner interface {
+	SignTransactions(txGroup []types.Transaction, indexesToSign []int) ([][]byte, []string, error)
```

Comment:
I'm leaning toward suggesting that this method should not return TxIDs as well. The reason being TxIDs can be derrived from the signed transactions, and if they're also returned we open the door to having inconsistent data (e.g. if a buggy signer returns a TxID that does not match the signed transaction).

What is this method returned `[]types.SignedTxn` instead of `[][]byte` and `[]string`? Then it would be easy to calculate the TxID for each signed transaction.